client/src/app/components/search/search.component.ts

```diff
@@ -36,10 +39,7 @@ export class SearchComponent implements OnInit {
         this.initializeSelectOptions();
     }
     public onDateSelection(date: NgbDate) {
-        let dateTemp = new Date(); 
-        dateTemp.setFullYear(date.year);
-        dateTemp.setMonth(date.month);
-        dateTemp.setDate(date.day);
+        let dateTemp = new Date(date.year, date.month-1, date.day+1); 
```

Comment:
Why you are adding +1 to day here ? Please replace to moment

Reply:
Switched to momentjs